pkgs/tools/text/ugrep/default.nix

```
@@ -0,0 +1,26 @@
+{ fetchFromGitHub, stdenv, pcre2, boost, zlib, lzham }:
+
+let
+  pname = "ugrep";
+  version = "2.4.1";
+in stdenv.mkDerivation {
+  pname = pname;
+  version = version;
+
+  src = fetchFromGitHub {
+    owner = "Genivia";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "16nr7zq4l2si9pfckfinbqnv94hw51z2qcbygc9x81jbjlvg3003";
+  };
+
+  buildInputs = [ pcre2 boost zlib lzham ];
+
+  meta = {
+    description = "Ultra fast grep with interactive query UI";
+    homepage = "https://github.com/Genivia/ugrep";
+    maintainers = with stdenv.lib.maintainers; [ numkem ];
+    license = stdenv.lib.licenses.bsd3;
+    platforms = stdenv.lib.platforms.all;
+  };
```

```suggestion
  meta = with stdenv.lib; {
    description = "Ultra fast grep with interactive query UI";
    homepage = "https://github.com/Genivia/ugrep";
    maintainers = with maintainers; [ numkem ];
    license = licenses.bsd3;
    platforms = platforms.all;
  };
```
this is also more readable

not tested. please do if accepted

---

@hmenke would it be ok to use 

```
let
  inherit (lib) maintainers licenses platforms;
in
...
```
maybe only in the meta section?

like you do in python

`from world import tree, river`

it looks really bad and is hard to write

```
stdenv.lib.licenses.bsd3;
stdenv.lib.platforms.all;
```

---

I don't know, I was just following https://nix.dev/anti-patterns/index.html